test/minigzip.c

```diff
@@ -33,8 +33,12 @@
 #endif
 
 #ifndef UNALIGNED_OK
```

Comment:
I am actually not sure why this is under UNALIGNED_OK check originally (I suspect it is checking the wrong thing), but I also think FreeBSD should not be in that category, at least not in all cases.
x86/x86_64 should define UNALIGNED_OK, probably also on FreeBSD if it does not already, and in that case this does not work like intended.

For this PR, I suggest adding a separate ifdef check for FreeBSD.
Then in a separate PR we can look into changing this whole block to behave the same way as in zutil_p.h.

Reply:
@dead2 Originally it was only used and required for PowerPC where UNALIGNED_OK is not defined.

Reply:
@mtl1979 Aha. That might not work great when you run Linux on ppc then.
In this case, we should probably try to detect the OS or feature instead to determine whether including malloc.h is needed.

Reply:
@Dead2 I haven't tried compiling the recent code on my Ubuntu/PPC, but I might when I have changed the hard disk to SSD drive...

Reply:
I agree, not sure why we even have `UNALIGNED_OK` check in here..

Reply:
if all "obsolete" memalign()s were switched to aligned_alloc() that "if __FreeBSD__" would not be necessary and we would only need to include stdlib.h instead of malloc.h for all or nearly all systems. else we would probably need to add __OpenBSD__ and the other BSDs also.

Reply:
Common sense says we should check in configure phase if aligned_alloc() is available and use it, instead of guessing which operating system (and version) and compiler supports it...

Reply:
@tpoechtrager Do you mind doing the changes suggested in the first comment above?

Reply:
@tpoechtrager *bump*

Reply:
> if all "obsolete" memalign()s were switched to aligned_alloc() that `if __FreeBSD__` would not be necessary and we would only need to include stdlib.h instead of malloc.h for all or nearly all systems. then again, its C11, so maybe no option for zlib-ng. anyway we probably also want to add `__OpenBSD__` and the other BSDs here, because they also dont know about malloc.h.

I looked into this, and unfortunately `aligned_alloc()` is C11. We target C99, so dropping the others might not be so easy.
But we could perhaps support `aligned_alloc()` in addition, and prefer it if it is available.
Any takers for making a PR that does this?

I don't know whether we can get FreeBSD and/or OpenBSD added to CI? I guess it would be useful, but don't know whether it is doable in GA.

Reply:
FreeBSD is not supported natively in GHA, but it seems some people have tried to do FreeBSD virtualization in macOS: https://github.com/vmactions/freebsd-vm.

Reply:
Also you can use https://builds.sr.ht or http://cirrus-ci.com for native CI